codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -528,6 +528,18 @@ private void generateHttpBindingTimestampSerializer(
         }
     }
 
+    private boolean isHttpDateTimestampHeader(Model model, HttpBinding.Location location, MemberShape memberShape) {
+        Shape targetShape = model.expectShape(memberShape.getTarget().toShapeId());
+        if (targetShape.getType() != ShapeType.TIMESTAMP) {
+            return false;
+        }
+
+        TimestampFormatTrait.Format format = model.getKnowledge(HttpBindingIndex.class).determineTimestampFormat(
+                memberShape, location, getDocumentTimestampFormat());
+
+        return format == Format.HTTP_DATE;
+    }
```

Comment:
Should this method also check for Binding Location as HEADER ?

Reply:
could rename this method to `isHttpDateTimestamp` instead. There isn't really anything Header specific about the internal behavior of this method.